scripts/package/fpsci_packager_generator.py

```diff
@@ -0,0 +1,125 @@
+# This script outputs a new version of `scripts/package/fpsci_packager.sh` based on the latest `data-files/log.txt`.
+# Both of those can be changed in the parameters below
+#
+# Once generated, `scripts/package/fpsci_packager.sh` should be run using bash from the root of the repository.
+# It will create a `dist/` directory and copy all of the needed files into it.
+#
+# The intention is that the contents of the `dist/` directory can then be zipped up for distribution.
+
+# Current release being built. Defaults to the git hash. Use "master", "branch name" or "tag name" for other builds.
+import subprocess
+currentRelease = subprocess.check_output(['git', 'rev-parse', '--short', 'HEAD']).strip().decode()
+# currentRelease = "master"
+
+# Name of the packaging script to generate
+outputScript = 'scripts/package/fpsci_packager.sh'
+
+# Path of the log.txt file to use
+inputLog = 'data-files/log.txt'
+
+# The list of files that should always be excluded
+excludeList = ['log.txt', 'keymap.Any', 'systemconfig.Any']
+# The list of files that should be excluded for release builds, but may want to be included for specific experiment distributions
+# Use the empty list if you want to distribute an experiment
+secondaryExcludeList = ['experimentconfig.Any', 'startupconfig.Any', 'userconfig.Any', 'userstatus.Any']
+# secondaryExcludeList = []
+
+# List of paths to search for to merge into the same distribution directory
+basePath = ['data-files', 'game', 'common']
+
+# Distribution path - where the output script will generate the distribution
+distPath = 'dist/'
+
+# Automatic configuation
+import os
+g3dPath = os.environ.get('g3d').replace('\\','/')
+runPath = os.getcwd().replace('\\','/')
+
+########## End script configuration ############
+
+from datetime import datetime
+now = datetime.now()
+
+if __name__ == '__main__':
+    # The log file which lists the files used
+    log = open(inputLog)
+
+    # Get the current set of lines in the script
+    lineSet = set(open(outputScript))
+    for line in sorted(lineSet):
+        if line.startswith('#'):
+            lineSet.remove(line)
+
+    # Hard coded files to copy
+    lineSet.add('mkdir -p ' + distPath + '\n')
+    lineSet.add('cp data-files/g3d-license.txt '  + distPath + '\n')
+    lineSet.add('cp LICENSE.txt '  + distPath + '\n')
+    lineSet.add('cp README.txt '  + distPath + '\n')
+    # For some reason these files aren't captured by G3D as used
+    lineSet.add('mkdir -p dist/shader/DefaultRenderer/\n')
+    lineSet.add('cp c:/g3d/G3D10/data-files/shader/DefaultRenderer/DefaultRenderer_OIT_writePixel.glsl dist/shader/DefaultRenderer/\n')
+    lineSet.add('mkdir -p dist/shader/UniversalSurface/\n')
+    lineSet.add('cp c:/g3d/G3D10/data-files/shader/UniversalSurface/UniversalSurface_depthPeel.pix dist/shader/UniversalSurface/\n')
+
+    # Set up copy of .exe
+    lineSet.add('cp Build/FirstPersonScience-x64-Release/FirstPersonScience.exe ' + distPath + '\n')
```

Comment:
Same as before, I prefer to keep this since G3D never emits this to a `log.txt`. This one in particular is the main executable, so we really want to hang on to it. Could potentially remove the others in the future if they end up being fixed in G3D.